Create support image via autoyast with default patterns on SLE 12 SP5

hjluo opened the pull request:
- Description:
   * Create support image via autoyast with default patterns on SLE 12 SP5.

- Related MR: N/A
- Test Cases: [**520**](https://openqa.suse.de/admin/job_templates/520?)
- Related ticket: [**132650**](https://progress.opensuse.org/issues/132650)
- Needles: N/A
- Verification run: 
   - [**Autoyast_VRs**](https://openqa.suse.de/tests/overview?arch=&flavor=&machine=&test=sle_autoyast_support_image_gnome_12sp5&modules=&module_re=&distri=sle&version=12-SP5&build=20230820&groupid=520#)
   - Migration
        * [**x86_64**](https://openqa.suse.de/tests/12047335 )
        * [**aarch64**](https://openqa.suse.de/tests/12047339)
        * [**s390x**](https://openqa.suse.de/tests/12047332)
        * [**ppc64le**](https://openqa.suse.de/tests/12047337)

lemon-suse commented:
I prefer to not select media installation, while to register scc to install with latest patches then de-register before publish qcow, and this will save time for patching in module combination jobs following.

hjluo commented:
> I prefer to not select media installation, while to register scc to install with latest patches then de-register before publish qcow, and this will save time for patching in module combination jobs following.

Changed and updated the VRs please take a look again. thanks.

chcao commented:
I think need to update the profile name by removing 'media'.

hjluo commented:
> I think need to update the profile name by removing 'media'.

Done && thank you!

lemon-suse commented on data/yam/autoyast/support_images/sles12sp5_install_default_pattern_aarch64.xml.ep:
```diff
@@ -0,0 +1,188 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Why still named as '.ep', I haven't seen the needed for it, please remove the extension name of 'ep' for all  the 4 files.

hjluo replied:
renamed.

lemon-suse commented on data/yam/autoyast/support_images/sles12sp5_install_default_pattern_aarch64.xml.ep:
```diff
@@ -0,0 +1,188 @@
+<profile xmlns="http://www.suse.com/1.0/yast2ns" xmlns:config="http://www.suse.com/1.0/configns">
+    <suse_register>
+        <do_registration config:type="boolean">true</do_registration>
+        <reg_code><%= $get_var->('SCC_REGCODE') %></reg_code>
```
Here should change to <reg_code>{{SCC_REGCODE}}</reg_code>

hjluo replied:
changed.

hjluo replied:
OK

lemon-suse commented on data/yam/autoyast/support_images/sles12sp5_install_default_pattern_aarch64.xml.ep:
```diff
@@ -0,0 +1,188 @@
+    </ntp-client>
+    <software>
+        <products config:type="list">
+            <product><%= uc $get_var->('SLE_PRODUCT') %></product>
```
change to {{SLE_PRODUCT}}

hjluo replied:
changed.

lemon-suse commented on data/yam/autoyast/support_images/sles12sp5_install_default_pattern_aarch64.xml:
```diff
@@ -0,0 +1,188 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
it should be default_patterns

hjluo replied:
done.

lemon-suse commented on data/yam/autoyast/support_images/sles12sp5_install_default_pattern_aarch64.xml:
```diff
@@ -0,0 +1,188 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
it should be default_patterns

lemon-suse commented:
PUBLISH_HDD_1: 'autoyast-SLES-%VERSION%-%ARCH%-media-default-updated.qcow2'

s/media/GM

lemon-suse commented:
Besides, need VRs of corresponding migration test to verify the qcow is ok.

rfan1 commented on data/yam/autoyast/support_images/sles12sp5_install_default_patterns_ppc64le.xml:
```diff
@@ -0,0 +1,121 @@
+      </interfaces>
+    </networking>
+    <firewall>
+      <enable_firewall config:type="boolean">true</enable_firewall>
```
Is it required for `qemu` backend？

rfan1 replied:
IMO, we need to open the ssh port for s390x platform only

hjluo replied:
keep it after we discuss this on slack. thanks

hjluo commented:
> Besides, need VRs of corresponding migration test to verify the qcow is ok.

OK, waiting in the queue.

hjluo commented:
@lemon-suse  @rfan1  now I've addressed all your comments and updated the latest VRs, please review it again, thanks.

rfan1 commented:
> @lemon-suse @rfan1 now I've addressed all your comments and updated the latest VRs, please review it again, thanks.

Seems you have defined `/home` as a subvolume on both x86-64 and ppc64le, however, based on the logs, /home is still shown as a separate partition. I am not sure why :)
Anyway, it didn't impact the test result, you can decide if you need to do some modifications [or just remove the `partitioning` part]

Others LGTM

hjluo commented:
> > @lemon-suse @rfan1 now I've addressed all your comments and updated the latest VRs, please review it again, thanks.
> 
> Seems you have defined `/home` as a subvolume on both x86-64 and ppc64le, however, based on the logs, /home is still shown as a separate partition. I am not sure why :) Anyway, it didn't impact the test result, you can decide if you need to do some modifications [or just remove the `partitioning` part]
removed home subvolume. thanks
> Others LGTM

lemon-suse commented:
LGTM, just the test case name of 'sle_autoyast_support_image_gnome_12sp5_def_patterns' should remove 'def_patterns', and qcow of 'autoyast-SLES-12-SP5-x86_64-GM-default-updated.qcow2' should remove 'default'. I will merge the PR, you still need one MR to update the qcow for corresponding migration test.

hjluo commented:
> LGTM, just the test case name of 'sle_autoyast_support_image_gnome_12sp5_def_patterns' should remove 'def_patterns', and qcow of 'autoyast-SLES-12-SP5-x86_64-GM-default-updated.qcow2' should remove 'default'. I will merge the PR, you still need one MR to update the qcow for corresponding migration test.

OK, I've changed the test cases and it's setting for your comments.
[**job_group_520**](https://openqa.suse.de/admin/job_templates/520?)

rfan1 reviewed (approved):
LGTM, we are safe to merge once the VRs are done
